### PR TITLE
✨ Mobile | Activity page v2024

### PR DESCRIPTION
**Pull request by zacharykeeping**
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

#562

> 2. What was changed?

Implements the new v2024 Activity page.

**Figure: New Activity page**

**Figure: Friends tab on new Activity page**

> 3. Did you do pair or mob programming?

No

### CODE REVIEW

File: src/Common/DTOs/ActivityFeed/ActivityFeedViewModel.cs
Comment (matt-goldman) [inline src/Common/DTOs/ActivityFeed/ActivityFeedViewModel.cs:1]: Please address the warnings in this file. We've been relaxed with warnings in existing code, but I don't want to introduce more.

File: src/MobileUI/Services/ActivityService.cs
Comment (matt-goldman) [inline src/MobileUI/Services/ActivityService.cs:1]: This service and abstraction should be removed:

1. The error handling should be done in the ViewModel
2. This demonstrates the lazy class code smell

File: src/Common/DTOs/ActivityFeed/ActivityFeedViewModel.cs
Comment (matt-goldman) [inline src/Common/DTOs/ActivityFeed/ActivityFeedViewModel.cs:1]: I think we should be consistent with naming. I think this should be named something like `ActivityFeedItemDto` (or something along those lines). We are typically using [...]ViewModel to represent a view, so I would expect this to contain a collection of `ActivityFeedItemDto` (or whatever it gets called) and some metadata.

**Review by matt-goldman — CHANGES_REQUESTED** [review]
Awesome work. I've noticed a few things that I think need some attention.

File: src/Common/DTOs/ActivityFeed/ActivityFeedViewModel.cs
Comment (zacharykeeping) [reply src/Common/DTOs/ActivityFeed/ActivityFeedViewModel.cs:1]: Done - the warnings have been addressed.

File: src/MobileUI/Services/ActivityService.cs
Comment (zacharykeeping) [reply src/MobileUI/Services/ActivityService.cs:1]: Done - this has been removed.

File: src/Common/DTOs/ActivityFeed/ActivityFeedViewModel.cs
Comment (zacharykeeping) [reply src/Common/DTOs/ActivityFeed/ActivityFeedViewModel.cs:1]: Done - the naming for this has been improved.

**Review by matt-goldman — APPROVED** [review]
LGTM 🚀 🚢 😎

**Pull request merged**